users/user_handler.go

```diff
@@ -0,0 +1,85 @@
+package users
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/SteveRusin/go_mentoring/randomId"
+)
+
+type RegisterUserDto struct {
+	UserName string `json:"userName"`
+	Password string `json:"password"`
+}
+
+type RegisterUserResponse struct {
+	Id       string `json:"id"`
+	UserName string `json:"userName"`
+}
+
+type LoginUserDto = RegisterUserDto
+
+type LoginUserResponse struct {
+	Url string `json:"url"`
+}
+
+var usersRepository = NewUserRepository()
+
+func HandleUser(w http.ResponseWriter, r *http.Request) {
```

Comment:
You can include this method in the user's structure, allowing you to incorporate usersRepository as a field within it for use in the handler

Alternatively, you can create a closure to encapsulate and manage the usersRepository variable.

Reply:
Thanks for the suggestion!
Let's leave it as is.